src/common/exif.cc

```diff
@@ -1378,6 +1378,7 @@ int dt_exif_read_blob(uint8_t *buf, const char *path, const int imgid, const int
       memcpy(buf + 6, &(blob[0]), length);
       return length + 6;
     }
+    fprintf(stderr, "[exif] Warning: too much data to store (%d bytes), Exif data will be discarded\n", length);
```

Comment:
There is no problems with reading exif, the problem is with writing it.
The warning should be in dt_exif_write_blob()